### PR TITLE
[codex] Fix hosted fresh install bootstrap

### LINKED ISSUE #2255: Hosted fresh install fails DB role init and admin bootstrap
## Summary

A clean hosted `install` on the alpha VPS exposed two greenfield deployment failures in the hosted HTTP integration branch. The public server can be made healthy manually, but the current one-liner path is not yet clean for a fresh deployment.

Tested branch/ref:

- `codex/hosted-http-integration`
- deployed revision: `eb023e8fd266c395c8da4cda8e8e3cd972f52e1b`
- alpha host: `mcp.dollhousemcp.com`

## What happened

After intentionally resetting the alpha runtime state while preserving `.env.production`, running:

```bash
scripts/hosted-remote-deploy.sh \
  --target root@178.156.247.192 \
  --hostname mcp.dollhousemcp.com \
  --public-base-url https://mcp.dollhousemcp.com \
  --deploy-dir /opt/dollhousemcp \
  --instance-name dollhousemcp \
  --mode cloud \
  --ref codex/hosted-http-integration \
  --skip-backup \
  install
```

built and migrated successfully, but first-run bootstrap was not clean:

1. Postgres did not create the `dollhouse_app` role on fresh `pgdata`.
2. `bootstrap-admin` failed when the deploy helper ran it in the production app image.
3. Running bootstrap in the maintenance image worked only after manually creating/granting the DB role and overriding `DOLLHOUSE_DATABASE_URL` to the admin DB URL.

## Evidence

Postgres log from fresh init:

```text
/docker-entrypoint-initdb.d/00-create-roles.sh: Permission denied
```

The rendered init script is mounted as:

```yaml
./init-db.sh:/docker-entrypoint-initdb.d/00-create-roles.sh:ro
```

Current renderer sets the host file to `0750`; inside the Postgres container the script was not readable/executable by the entrypoint user.

Bootstrap in the production app image failed with:

```text
Error [ERR_MODULE_NOT_FOUND]: Cannot find package 'commander' imported from /app/dist/cli/admin-bootstrap.js
```

Bootstrap in the maintenance image then failed until the DB role was fixed and the admin URL was supplied:

```text
Failed to mark bootstrap complete: withSystemContext requires a PostgreSQL role with SUPERUSER or BYPASSRLS; current_user=dollhouse_app
```

## Manual alpha workaround used

- Created `dollhouse_app` manually using the current `POSTGRES_PASSWORD` value.
- Applied `post-migration-grants.sql` manually.
- Ran `dollhouse-admin-bootstrap` from `dollhousemcp-migrate` with `DOLLHOUSE_DATABASE_URL` set to `DOLLHOUSE_DATABASE_ADMIN_URL`.
- Restarted `dollhousemcp` and `caddy`.

After that, verification passed:

- `/healthz` => 200
- `/readyz` => 200
- unauthenticated `/mcp` => 401

## Suggested fixes

- Make `init-db.sh` readable/executable in the Postgres container, or mount it in a shape the official Postgres entrypoint can reliably source/execute.
- Run post-migration grants as part of the hosted deploy flow after migrations, so app role access is correct on fresh DBs.
- Change hosted `bootstrap-admin` to run from the builder/maintenance image, or include required CLI runtime deps in the production image.
- Ensure bootstrap CLI uses the admin DB URL, or teach the deploy helper to pass `DOLLHOUSE_DATABASE_URL=$DOLLHOUSE_DATABASE_ADMIN_URL` for `bootstrap-admin`.
- Add a fresh-install integration test that starts from empty `pgdata` and verifies app role creation, admin bootstrap, `/readyz`, and unauthenticated `/mcp`.

## Acceptance criteria

- Clean hosted install from empty runtime state succeeds without manual DB edits.
- `scripts/hosted-remote-deploy.sh ... install` can bootstrap or clearly report the required next operator command.
- `scripts/hosted-remote-deploy.sh ... bootstrap-admin` succeeds without dependency/runtime image errors.
- Fresh install verification passes without hand-created roles or grants.

### PR DESCRIPTION
**Pull request by mickdarling**
## Summary

Fixes #2255.

This repairs the hosted fresh-install path that failed when alpha was reset to an empty runtime state:

- renders the Postgres init script as `0755` so the official Postgres entrypoint can read/execute the bind-mounted role setup script
- renders a `post-migration-grants.sql` file and applies it after every migration pass so fresh installs and updates converge on the same `dollhouse_app` privileges
- runs `bootstrap-admin` from the maintenance/migrate image instead of the production app image
- forces bootstrap to use `DOLLHOUSE_DATABASE_ADMIN_URL` as `DOLLHOUSE_DATABASE_URL`, which keeps the RLS bootstrap path in the admin DB context
- expands hosted deploy tests to cover the grant pass, file modes, dry-run messaging, and bootstrap command shape

## Verification

- `npm run test:hosted-deploy`
- `npm run lint:shell`
- `git diff --check`
- `bash -n scripts/hosted-deploy.sh scripts/hosted-deploy/*.sh tests/scripts/hosted-deploy-render.test.sh tests/scripts/hosted-deploy-workflow.test.sh tests/scripts/hosted-remote-deploy.test.sh`

Build note:

- `npm run build` is blocked in this local checkout because `tsc` is not installed in `node_modules` (`sh: tsc: command not found`).

### CODE REVIEW

**Comment by mickdarling** [comment]
Addressed the review suggestions in `15d731fa`:

- Replaced the nested `compose exec ... sh -c ... psql` grant command with a rendered `/usr/local/bin/apply-post-migration-grants` helper mounted into the Postgres container.
- Replaced the bootstrap `sh -c` command with a rendered `/usr/local/bin/dollhouse-bootstrap-admin` helper mounted into the maintenance container. The admin DB URL stays inside the container environment and is not passed through a shell command string.
- Added render-time validation that generated runtime files are readable/executable as required.
- Expanded dry-run output to call out the generated helper scripts and secret-handling shape.
- Extended tests to assert helper mounts, file modes, secret non-embedding, dry-run messaging, and the simplified Docker command shape.

Local verification after the follow-up:

- `bash -n scripts/hosted-deploy.sh scripts/hosted-deploy/*.sh tests/scripts/hosted-deploy-render.test.sh tests/scripts/hosted-deploy-workflow.test.sh tests/scripts/hosted-remote-deploy.test.sh`
- `npm run test:hosted-deploy`
- `npm run lint:shell`
- `git diff --check`

The remaining suggestion for actual Postgres entrypoint/container-start integration coverage is still better handled by the existing broader integration-test follow-up rather than bloating this fix PR.

File: scripts/hosted-deploy/runtime.sh [hunk 1]
```diff
@@ -130,7 +133,9 @@ run_bootstrap_admin() {
   fi
 
   log "bootstrapping GitHub admin"
-  compose run --rm dollhousemcp npx dollhouse-admin-bootstrap "${args[@]}"
+  compose run --rm \
+    dollhousemcp-migrate \
```
[the remark this answers is not in the document]
Comment (mickdarling) [reply scripts/hosted-deploy/runtime.sh:139]: Fixed in `0c2824fa` by setting `DOLLHOUSE_AUTH_STORAGE_BACKEND: postgres` on the `dollhousemcp-migrate` service as well as the app service. Added a render test assertion that the generated compose file contains the Postgres auth backend twice, so the bootstrap CLI keeps using the same durable auth storage as the hosted AS.
